sklearn/impute/_base.py

```diff
@@ -75,61 +75,27 @@ def __init__(self, missing_values=np.nan, add_indicator=False):
         self.missing_values = missing_values
         self.add_indicator = add_indicator
 
-    @abstractmethod
-    def _fit(self, X, y=None):
-        """Specialized fit method implemented by each imputer."""
-
-    def fit(self, X, y=None):
-        """Fit the imputer on X.
-
-        Parameters
-        ----------
-        X : array-like shape of (n_samples, n_features)
-            Input data, where `n_samples` is the number of samples and
-            `n_features` is the number of features.
-
-        Returns
-        -------
-        self : object
-        """
+    def _fit_indicator(self, X):
+        """Fit a MissingIndicator if required."""
         if self.add_indicator:
+            self._copy = True
             self.indicator_ = MissingIndicator(
-                missing_values=self.missing_values, error_on_new=False)
+                missing_values=self.missing_values, error_on_new=False
+            )
             self.indicator_.fit(X)
         else:
+            self._copy = getattr(self, 'copy', True)
             self.indicator_ = None
-        self._fit(X, y)
-        return self
 
     @abstractmethod
-    def _transform(self, X):
-        """Specialized transform method implemented by each imputer."""
-
-    def _concatenate_indicator(self, X, X_trans_imputer):
-        """Concatenate indicator if required"""
+    def _transform_indicator(self, X, X_trans_imputer):
```

Comment:
This does the transform and the concatenate. An extra explicit name would be `_transform_concat_indicator`.